scenarios/kubernetes_e2e.py

```diff
@@ -44,49 +45,169 @@ def check(*cmd):
     subprocess.check_call(cmd)
 
 
-def sig_handler(_signo, _frame):
-    """Stops container upon receive signal.SIGTERM and signal.SIGINT."""
-    print >>sys.stderr, 'signo = %s, frame = %s' % (_signo, _frame)
-    check('docker', 'stop', CONTAINER)
+def check_env(env, *cmd):
+    """Log and run the command with a specific env, raising on errors."""
+    print >>sys.stderr, 'Environment:'
+    for key, value in env.items():
+        print >>sys.stderr, '%s=%s' % (key, value)
+    print >>sys.stderr, 'Run:', cmd
+    subprocess.check_call(cmd, env=env)
 
 
 def kubekins(tag):
     """Return full path to kubekins-e2e:tag."""
     return 'gcr.io/k8s-testimages/kubekins-e2e:%s' % tag
 
 
-def main(args):
-    """Set up env, start kubekins-e2e, handle termination. """
-    # pylint: disable=too-many-locals
-
-    # dockerized-e2e-runner goodies setup
-    workspace = os.environ.get('WORKSPACE', os.getcwd())
-    artifacts = '%s/_artifacts' % workspace
-    if not os.path.isdir(artifacts):
-        os.makedirs(artifacts)
+class LocalMode(object):
+    """Runs e2e tests by calling e2e-runner.sh."""
+    def __init__(self, workspace):
+        self.workspace = workspace
+        self.env = []
+        self.env_files = []
+        self.add_environment(
+            'HOME=%s' % workspace,
+            'WORKSPACE=%s' % workspace,
+            'PATH=%s' % os.getenv('PATH'),
+        )
+
+    @staticmethod
+    def parse_env(env):
+        """Returns (FOO, BAR=MORE) for FOO=BAR=MORE."""
+        return env.split('=', 1)
+
+    def add_environment(self, *envs):
+        """Adds FOO=BAR to the list of environment overrides."""
+        self.env.extend(self.parse_env(e) for e in envs)
+
+    def add_files(self, env_files):
+        """Reads all FOO=BAR lines from each path in env_files seq."""
+        for env_file in env_files:
+            with open(test_infra(env_file)) as fp:
+                for line in fp:
+                    line = line.rstrip()
+                    if not line or line.startswith('#'):
+                        continue
+                    self.env_files.append(self.parse_env(line))
+
+    def add_gce_ssh(self, priv, pub):
+        """Copies priv, pub keys to $WORKSPACE/.ssh."""
+        gce_ssh = '%s/.ssh/google_compute_engine' % self.workspace
+        gce_pub = '%s/.ssh/google_compute_engine.pub' % self.workspace
+        shutil.copy(priv, gce_ssh)
+        shutil.copy(pub, gce_pub)
+        self.add_environment(
+            'JENKINS_GCE_SSH_PRIVATE_KEY_FILE=%s' % gce_ssh,
+            'JENKINS_GCE_SSH_PUBLIC_KEY_FILE=%s' % gce_pub,
+        )
+
+    def add_service_account(self, path):
+        """Sets GOOGLE_APPLICATION_CREDENTIALS to path."""
+        self.add_environment('GOOGLE_APPLICATION_CREDENTIALS=%s' % path)
+
+    @property
+    def runner(self):
+        """Finds the best version of e2e-runner.sh."""
+        options = ['e2e-runner.sh', test_infra('jenkins/e2e-image/e2e-runner.sh')]
```

Comment:
I'm depending on the WORKDIR line: https://github.com/kubernetes/test-infra/blob/master/jenkins/e2e-image/Dockerfile#L68